src/FormCollection/README.md

```diff
@@ -0,0 +1,190 @@
+in your `package.json` file.
+
+## Use predefined theme
+
+You need to select the right theme from the one you are using:
+
+```yaml
+# config/packages/twig.yaml
+twig:
+    # For bootstrap for example
+    form_themes: ['@FormCollection/form_theme_div.html.twig']
+```
+
+There are 2 predefined themes available:
+
+-   `@FormCollection/form_theme_div.html.twig`
+-   `@FormCollection/form_theme_table.html.twig`
+
+[Check the Symfony doc](https://symfony.com/doc/4.4/form/form_themes.html) for the different ways to set themes in Symfony.
+
+## Use a custom form theme
+
+Consider your `BlogFormType` form set up and with a comments field that is a `CollectionType`, you can
+render it in your template:
+
+```twig
+{% macro commentFormRow(commentForm) %}
+    <div
+        class="col-4"
+        data-symfony--ux-form-collection--collection-target="entry"
+    >
+        {{ form_errors(commentForm) }}
+        {{ form_row(commentForm.content) }}
+        {{ form_row(commentForm.otherField) }}
+
+        <button type="button" data-action="symfony--ux-form-collection--collection#delete">
+            Remove
+        </button>
+    </div>
+{% endmacro %}
+
+<div
+    class="row"
+    data-controller="symfony--ux-form-collection--collection"
+    data-prototype="{{ _self.commentFormRow(form.comments.vars.prototype)|e }}"
+>
+    {% for commentForm in form.comments %}
+        {{ _self.commentFormRow(commentForm) }}
+    {% endfor %}
+
+    <button type="button" data-action="symfony--ux-form-collection--collection#add">
+        Add Another
+    </button>
+</div>
+```
+
+## Usage
+
+The most common usage of Form Collection is to use it as a replacement of
+the native CollectionType class:
+
+```php
+// ...
+use Symfony\UX\FormCollection\Form\UXCollectionType;
+
+class BlogFormType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            // ...
+            ->add('comments', UXCollectionType::class, [
+                // ...
+                'button_add_options' => [
+                    'label' => 'Add',                       // Default text for the add button (used by predefined theme)
+                    'class' => 'btn btn-outline-primary',   // Add HTML classes to the add button (used by predefined theme)
+                ],
+                'button_delete_options' => [
+                    'label' => 'Remove',                    // Default text for the delete button (used by predefined theme)
+                    'class' => 'btn btn-outline-secondary', // Add HTML classes to the add button (used by predefined theme)
+                ],
+            ])
+            // ...
+        ;
+    }
+
+    // ...
+}
+```
+
+You can display it using Twig as you would normally with any form:
+
+```twig
+{# edit.html.twig #}
+
+{{ form_start(form) }}
+    {# ... #}
+    {{ form_row(comments) }}
+    {# ... #}
+{{ form_end(form) }}
+```
```

Comment:
Somewhere, we need to mention that, in order for this to work, they likely need `orphanRemoval` and `cascade=['persist']` on their entities. This is what my field looks like:

```php
#[ORM\OneToMany(mappedBy: 'directory', targetEntity: File::class, orphanRemoval: true, cascade: ['persist'])]
private $files;
```